mathesar_ui/src/systems/data-explorer/urlSerializationUtils.ts

```diff
@@ -34,56 +47,68 @@ export function constructDataExplorerUrlToSummarizeFromGroup(
   return undefined;
 }
 
-export function constructQueryModelFromTerseSummarizationHash(
+export function constructQueryModelFromHash(
   hash: string,
 ): UnsavedQueryInstance {
+  const queryInstance: Mutable<UnsavedQueryInstance> = {};
   const terseSummarization = JSON.parse(
     Url64.decode(hash),
-  ) as TerseSummarization;
-  const groupedColumnId = terseSummarization.terseGrouping[0][0];
-  const groupedColumn = terseSummarization.columns.find(
-    (column) => column.id === groupedColumnId,
-  );
-  if (!groupedColumn) {
-    return {};
+  ) as Partial<TerseSummarization>;
+
+  if (terseSummarization.baseTableId) {
+    queryInstance.base_table = terseSummarization.baseTableId;
```

Comment:
This mutation, and the addition/usage of the new `Mutable` type seems fishy to me. `QueryInstance` defines all its parameters to be `readonly` and yet we mutating it, essentially using the new `Mutable` type to disregard those `readonly` specifiers.

I'd like for @pavish to look at this before we merge.

I haven't spent enough time looking at this file to understand what it does, and I'm not familiar with our design and data structures for the Data Explorer. But my knee jerk reaction is: "there must be a cleaner way to do this".